### PR TITLE
Fix BCC recipients visible in sent message

### LINKED ISSUE #1938: BCC not working as expected
### Expected behavior
I wrote the mail with BBC on my Android 6.0.1.
So other people from CC shouldn't know that there are e-mail on BBC

### Actual behavior
All were able to read that there was bbc

### Steps to reproduce
1. Write mail with BBC
2. See their smiling faces when they discovers that boss e-mail is on BBC.

### Environment
K-9 Mail version: 5.201

Android version:6.0.1.

Account type (IMAP, POP3, WebDAV/Exchange): IMAP and SMTP on zoho.com

### PR DESCRIPTION
**Pull request by cketti**
Looks like this was broken in all alpha releases since January 2015 -> https://github.com/k9mail/k-9/commit/523ebd0f2afeba538924a5836bc242429c8202ed

This whole mutable `LocalMessage` thing is a mess. I feel dirty :(

Fixes #1938

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java [hunk 1]
```diff
@@ -40,6 +42,7 @@
     private long messagePartId;
     private String mimeType;
     private PreviewType previewType;
+    private boolean dirty = false;
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java]: Please lets call it `headersNeedUpdating`

**Comment by Valodim** [comment]
Ooh, ouch. No tests though? Couldn't think of a good way to pin the behavior?

File: k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java (hunk 1 above)
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java]: :+1: Fixed

**Comment by cketti** [comment]
@Valodim: We should probably do that eventually. For now I'd like to get the fix out as fast as possible.

**Pull request merged**